Seda/Seda/Crypto.swift

```diff
@@ -0,0 +1,216 @@
+    var pub_key: SecKey? = nil
+    
+    init() {
+        guard let tag = self.keyTag_raw else {
+            print("SOMETHING IS VERY WRONG: Can't unwrap a standard string in Crypto initializer")
+            // The keyTag needs to be initialized in the event that this cannot be unwrapped
+            // Something would have to be very wrong for us to end up here
+            let bytesPointer = UnsafeMutableRawPointer.allocate(byteCount: 1, alignment: 1)
+            self.keyTag = Data(bytes: bytesPointer, count: 1)
+            return
+        }
+
+        self.keyTag = tag
+        
+        // Load the private key, if it doesn't exist, create a new one
+        loadPrivKey()
+        if priv_key == nil {
+            createPrivKey() // Create the private key
+        }
+    }
+    
+    
+    // Creates a private key and stores it in the enclave
+    func createPrivKey() {
+        let attributes: [String: Any] = [
+            kSecAttrKeyType         as String:   keyType,
+            kSecAttrKeySizeInBits   as String:   keySize,
+            kSecAttrTokenID         as String:   kSecAttrTokenIDSecureEnclave, // Store the key in the enclave
+            kSecPrivateKeyAttrs     as String: [
+                kSecAttrIsPermanent     as String: true,   // Store the key in the keychain, so we will be able to get the key when the user logs back into the application
+                kSecAttrApplicationTag  as String : keyTag
+                
+            ]
+        ]
+        
+        var err: Unmanaged<CFError>?
+        guard let privateKey = SecKeyCreateRandomKey(attributes as CFDictionary, &err) else {
+            print("Error in creating the private key" + (err!.takeRetainedValue() as! String))
+            return
+        }
+        
+        priv_key = privateKey
+    }
+    
+    // Creates a public key
+    func createPublicKey() -> SecKey? {
+        guard let key = priv_key else {
+            print("The user does not have a private key")
+            return nil
+        }
+        
+        guard let publicKey = SecKeyCopyPublicKey(key) else {
+            print("Unable to generate a public key")
+            return nil
+        }
+        
+        pub_key = publicKey
+        return publicKey
+    }
+    
+    /*
+     1. Check to see if private key exists
+     2. If private key does not exist create one
+     */
+    func loadPrivKey() {
+        // Make a query to retrieve the key from the keychain
+        let query: [String: Any] = [
+            kSecClass               as String: kSecClassKey,
+            kSecAttrApplicationTag  as String: keyTag,
+            kSecAttrKeyType         as String: keyType,
+            kSecReturnRef           as String: true
+        ]
+        
+        var item: CFTypeRef?
+        let status = SecItemCopyMatching(query as CFDictionary, &item)
+        guard status == errSecSuccess else {
+            print("Crypto.loadPrivKey(): Error in retrieving the private key")
```

Comment:
While I get the above error (EXC_BAD_ACCESS), this line is printed to the console.